core/src/main/java/hudson/model/AutoCompletionCandidates.java

```diff
@@ -127,8 +128,10 @@ public void onItem(Item i) {
                  && i.hasPermission(Item.READ)
                     // and read permission required
                 ) {
-                    if (type.isInstance(i) && n.startsWith(value))
+                    if (type.isInstance(i) && n.startsWith(value) &&
+                            (!(container instanceof DirectlyModifiableTopLevelItemGroup) || !(i instanceof TopLevelItem) || ((DirectlyModifiableTopLevelItemGroup) container).canAdd((TopLevelItem) i))) {
```

Comment:
Probably not applicable, since that is generally used for descriptors—new item types—whereas this would be used for copying existing items, for which `canAdd` is more specific since it is passed an actual instance.

At any rate, I am going to :bug: myself since I realized that this method may well be called by things that have nothing to do with copying items; would be better to do it from `ViewDescriptor.doAutoCompleteCopyNewItemFrom`.